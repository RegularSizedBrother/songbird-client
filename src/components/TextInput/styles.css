.test > i {
  position: absolute;
  display: block;
  transform: translate(0, -52%);
  top: 50%;
  pointer-events: none;
  width: 37px;
  text-align: center;
  font-style: normal;
  z-index: 9;
  color: rgb(100, 100, 100);
}

.test > input {
  padding-left: 25px;
  padding-right: 0;
  /* border-radius: 10px; */
}
