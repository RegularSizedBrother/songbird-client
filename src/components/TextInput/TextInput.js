import React from 'react';
import './styles.css';

function TextInput(props) {
  return(
    <form onSubmit={props.onSubmit}>
      <div className='form-group'>
        <div className='input-group test'>
            <input type='text' className='form-control' placeholder={props.placeholder} value={props.value} onChange={props.onChange} />
            <i>@</i>
        </div>
        <label className='text-light ml-4 mt-1'>Enter your twitter handle...</label>
      </div>
    </form>
  );
}

export default TextInput;
