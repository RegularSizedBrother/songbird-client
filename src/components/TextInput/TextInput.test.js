import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TextInput from './TextInput';

describe('TextInput', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      onSubmit: jest.fn(() => 'onSubmit'),
      onChange: jest.fn(() => 'onChange'),
      placeholder: 'BarackObama',
      value: '',
    };
    wrapper = shallow(<TextInput {...props} />);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<TextInput {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  it('calls onSubmit when the form is submitted', () => {
    const form = wrapper.find('form').first();
    form.simulate('submit');
    expect(props.onSubmit).toBeCalled();
  });

  it('calls onChange when the input changes value', () => {
    const input = wrapper.find('input').first();
    input.simulate('change', { target: { value: 'OSUPrezDrake' } });
    expect(props.onChange).toBeCalled();
  });
});
