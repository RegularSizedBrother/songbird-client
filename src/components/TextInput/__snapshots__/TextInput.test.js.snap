// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <div
      className="input-group test"
    >
      <input
        className="form-control"
        type="text"
      />
      <i>
        @
      </i>
    </div>
    <label
      className="text-light ml-4 mt-1"
    >
      Enter your twitter handle...
    </label>
  </div>
</form>
`;
